cmake/Utils.cmake

```diff
@@ -104,7 +104,9 @@ function(xgboost_set_cuda_flags target)
     target_compile_definitions(${target} PRIVATE -DXGBOOST_USE_NVTX=1)
   endif()
 
-  target_link_libraries(${target} PRIVATE CCCL::CCCL)
+  target_link_libraries(${target}
+    INTERFACE CCCL::CCCL
```

Comment:
Does this propagate into the C interface?

Reply:
My bad. A quick experiment with the C API demo reveals that the installed CMake config "leaks" CCCL requirement to the consumer of libxgboost. Let me rectify this.